scene/2d/tile_map.h

```diff
@@ -108,62 +157,170 @@ struct TileMapQuadrant {
 	}
 };
 
-class TileMap : public Node2D {
-	GDCLASS(TileMap, Node2D);
-
+class TileMapLayer : public RefCounted {
 public:
-	class TerrainConstraint {
-	private:
-		const TileMap *tile_map;
-		Vector2i base_cell_coords;
-		int bit = -1;
-		int terrain = -1;
-
-		int priority = 1;
-
-	public:
-		bool operator<(const TerrainConstraint &p_other) const {
-			if (base_cell_coords == p_other.base_cell_coords) {
-				return bit < p_other.bit;
-			}
-			return base_cell_coords < p_other.base_cell_coords;
-		}
+	enum DataFormat {
+		FORMAT_1 = 0,
```

Comment:
```suggestion
		FORMAT_1,
```
This is default.

Reply:
I think I'd prefer to keep it that way. Since this enum is serialized I thinks it's a bit better to be explicit about the starting value (especially since `FORMAT_1 = 0` and not `FORMAT_1 = 1`.